lib/vast/Frontend/GenAction.cpp

```diff
@@ -132,33 +145,88 @@ namespace vast::cc {
             );
         }
 
+        // TODO: Introduce helper wrapper on top of `vast_args`?
         enum class target_dialect {
             high_level
+            , low_level
+            , llvm
         };
 
+        target_dialect parse_target_dialect(const vast_args::maybe_option_list &list)
+        {
+            if (!list)
+                return target_dialect::high_level;
+
+            if (list->size() != 1)
+                VAST_UNREACHABLE("Can emit only one dialect.");
+
+            return parse_target_dialect(list->front());
+        }
+
+        target_dialect parse_target_dialect(llvm::StringRef from)
+        {
+            auto trg = from.lower();
+            if (trg == "hl" || trg == "high_level")
```

Comment:
we should add some single place description of dialects and use named constants or some other tag-dispatch-like mechanism, now it is starting to be bunch or random string all over the place